scripts/object_storage_cleaner/storage_cleaner.py

```diff
@@ -0,0 +1,628 @@
+                        "Version: Key=%s VersionId=%s Size=%s LastModified=%s",
+                        v.key,
+                        v.version_id,
+                        v.size,
+                        v.last_modified,
+                    )
+
+        self._print_summary(prune=False)
+
+    def prune(self) -> None:
+        """Scan and delete logically deleted objects."""
+
+        self._print_filters()
+
+        batch: list[dict[str, str]] = []
+
+        def progress(delta: int) -> None:
+            self._print_progress(delta, label="Deleted")
+
+        logger.debug("Starting to delete logically deleted objects...")
+
+        # Iterate
+        for aggregate, versions in self._iter_logically_deleted(progress=progress):
+            self._stats_keys += 1
+            self._stats_bytes += aggregate.total_size_bytes
+
+            for v in versions:
+                batch.append({"Key": v.key, "VersionId": v.version_id})
+                if len(batch) >= self.BATCH_SIZE:
+                    self._stats_versions += self._delete_batch(batch)
+                    batch.clear()
+                    self._print_progress(0, label="Deleted")
+
+        # Flush remaining
+        if batch:
+            self._stats_versions += self._delete_batch(batch)
+
+        self._print_summary(prune=True)
+
+
+def _handle_sigint(sig: int, frame: object) -> None:
+    logger.info("\n\nScan interrupted by user.")
+    sys.exit(0)
+
+
+signal.signal(signal.SIGINT, _handle_sigint)
+
+
+def setup_logging(debug: bool = False, log_file: str | None = None) -> None:
+    """Setup logging for the script."""
+    level = TRACE_LEVEL if log_file else (logging.DEBUG if debug else logging.INFO)
+    logger.setLevel(level)
+    logger.handlers.clear()
+    logger.propagate = False
+
+    # 2. Console Handler (The User UI)
+    console_handler = logging.StreamHandler(sys.stdout)
+    console_handler.setLevel(logging.DEBUG if debug else logging.INFO)
+    console_handler.setFormatter(logging.Formatter("%(message)s"))
+
+    # Filter out errors (they go to stderr)
+    console_handler.addFilter(lambda record: record.levelno <= logging.INFO)
+    logger.addHandler(console_handler)
+
+    # 3. Stderr Handler (Errors)
+    error_handler = logging.StreamHandler(sys.stderr)
+    error_handler.setLevel(logging.WARNING)
+    error_handler.setFormatter(logging.Formatter("%(levelname)s: %(message)s"))
+    logger.addHandler(error_handler)
+
+    # 4. File Handler (The Audit Log)
+    if log_file:
+        Path(log_file).parent.mkdir(parents=True, exist_ok=True)
+        file_handler = RotatingFileHandler(
+            log_file, maxBytes=1000 * 1024 * 1024, backupCount=5, encoding="utf-8"
+        )
+        file_handler.setLevel(TRACE_LEVEL)
+        file_handler.setFormatter(
+            logging.Formatter(
+                "%(asctime)s [%(levelname)s] %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
+            )
+        )
+        logger.addHandler(file_handler)
```

Comment:
Leave this to the outer level, there are log rotators that are doing this much better than this code. Just kill this functionality.

Reply:
One run of the script should produce one log file. We can easily do that with a timestamp suffix.

If we want cleanup or something more advanced, our good friend https://linux.die.net/man/8/logrotate exists.